jwt/api.py

```diff
@@ -110,14 +129,16 @@ def encode(self, payload, key, algorithm='HS256', headers=None, json_encoder=Non
 
         return b'.'.join(segments)
 
-    def decode(self, jwt, key='', verify=True, algorithms=None, **kwargs):
+    def decode(self, jwt, key='', verify=True, algorithms=None, options=None, **kwargs):
         payload, signing_input, header, signature = self._load(jwt)
 
         if verify:
-            self._verify_signature(payload, signing_input, header, signature,
-                                   key, algorithms)
+            merged_options = self._merge_options(override_options=options)
+            if merged_options.get('verify_signature'):
+                self._verify_signature(payload, signing_input, header, signature,
+                                       key, algorithms)
```

Comment:
> Adding an options keyword argument that can take five different keys is the same as adding those keys to the method interface

I disagree with you here entirely. The options object is much more readable than simple adding an argument to the method's arguments and that gives it an advantage over simply adding arguments.

I think maintaining some level of parity with ruby-jwt's API is also valuable here. (even though I'm not saying we should lock ourselves down to just matching ruby-jwt either; its just that I happen to agree with their design)